src/cloud-cli/cli.ts

```diff
@@ -49,23 +49,37 @@ program
     }
   });
 
-/*** APPLICATIONS MANAGEMENT ***/
+/////////////////////////////
+/* APPLICATIONS MANAGEMENT */
+/////////////////////////////
+
 const applicationCommands = program
   .command('applications')
   .description('Manage your DBOS applications')
   .option('-h, --host <string>', 'Specify the host', DEFAULT_HOST)
-  .option('-p, --port <port>', 'Specify the port', DEFAULT_PORT)
+  .option('-p, --port <string>', 'Specify the port', DEFAULT_PORT)
 
 applicationCommands
   .command('register')
   .description('Register a new application')
   .requiredOption('-n, --name <string>', 'Specify the app name')
-  .option('-m, --machines <number>', 'Number of VMs to deploy', '1')
+  .option('-m, --machines <string>', 'Number of VMs to deploy', '1')
   .action(async (options: { name: string, machines: string }) => {
     const { host, port }: { host: string, port: string } = applicationCommands.opts()
     await registerApp(options.name, host, port, parseInt(options.machines));
   });
 
+applicationCommands
```

Comment:
Can we merge update with deploy? For example, deploy will become update if the app has been deployed before.

Reply:
Or rather, does update only work for already deployed apps? Or can we also update registered but not deployed apps as well? What's the semantics here?

Reply:
I have been thinking about this for a while and am not yet convinced we should do this merge.

The reason is that deploy is, really, a glorified "control + S" shortcut dedicated to updating application code / version. It should be fast and not cluttered with whatever updating the record may do - in the future it will increase in scope.

Conversely, merging the two would force users to deploy code each time they want to change another parameter, say, `max_vms`.

Another reason is that deploy require the request to be a multiparts/form upload, which forces an entirely different parameter processing/validation logic than our other endpoints which use JSON bodies.

So, as it stands, I don't think we should try merging the two. There might be a killer use-case, but I can't see it today.

Reply:
> Or rather, does update only work for already deployed apps? Or can we also update registered but not deployed apps as well? What's the semantics here?

I saw this after posting my previous comment. Update works for all registered apps. So you can update an app that does not have had code deployed yet. This makes me think that, because update also triggers a host<->VM assignment, if you do an update before the deploy, resolvers will start attempting starting VMs, which is not ideal. But I don't think this is a big problem for now.

Reply:
Agree with Max, simpler to separate them